airflow/models/dagrun.py

```diff
@@ -649,19 +651,41 @@ def task_instance_scheduling_decisions(self, session: Session = NEW_SESSION) ->
 
     def _get_ready_tis(
         self,
-        scheduleable_tasks: List[TI],
+        scheduleable_tis: List[TI],
         finished_tis: List[TI],
         session: Session,
     ) -> Tuple[List[TI], bool]:
         old_states = {}
         ready_tis: List[TI] = []
         changed_tis = False
 
-        if not scheduleable_tasks:
+        if not scheduleable_tis:
             return ready_tis, changed_tis
 
+        # If we expand TIs, we need a new list so that we iterate over them too. (We can't alter
+        # `scheduleable_tis` in place and have the `for` loop pick them up
+        expanded_tis: List[TI] = []
+
         # Check dependencies
-        for st in scheduleable_tasks:
+        for st in itertools.chain(scheduleable_tis, expanded_tis):
```

Comment:
Maybe call this `sti` instead to signify it’s a ti, not a t(ask)? (or `schedulable_ti`)

Reply:
Ah, Shimano Total Integration. Yeah, best to avoid that.